examples/shader/custom_render_phase.rs

```diff
@@ -0,0 +1,505 @@
+            // that data and you can only do it in the ExtractSchedule.
+            // Some common extract scenarios have plugins that will do this automatically.
+            // For the purpose of the example we will do it manually.
+            .add_systems(ExtractSchedule, extract_render_phase)
+            .add_systems(
+                Render,
+                (
+                    queue_mesh_custom_phase.in_set(RenderSet::Queue),
+                    // This will automatically sort all items in the phase based on the [`PhaseItem::sort_key()`]
+                    sort_phase_system::<CustomPhaseItem>.in_set(RenderSet::PhaseSort),
+                    prepare_custom_bind_group.in_set(RenderSet::PrepareBindGroups),
+                ),
+            );
+
+        // Bevy's renderer uses a render graph which is a collection of nodes in a directed acyclic graph.
+        // It currently runs on each view/camera and executes each node in the specified order.
+        // It will make sure that any node that needs a dependency from another node
+        // only runs when that dependency is done.
+        //
+        // Each node can execute arbitrary work, but it generally runs at least one render pass.
+        // A node only has access to the render world, so if you need data from the main world
+        // you need to extract it manually or with the plugin like above.
+        render_app
+            // Add the node that will render the custom phase
+            .add_render_graph_node::<ViewNodeRunner<CustomNode>>(CORE_3D, CustomNode::NAME)
+            // This will schedule the custom node to run after the main opaque pass
+            .add_render_graph_edge(
+                CORE_3D,
+                core_3d::graph::node::MAIN_OPAQUE_PASS,
+                CustomNode::NAME,
+            );
+    }
+
+    fn finish(&self, app: &mut App) {
+        // We need to get the render app from the main app
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+        // This needs to be after the initial build because it needs a reference to the RenderDevice
+        // but it doesn't exist in the build() step
+        render_app.init_resource::<CustomPipeline>();
+    }
+}
+
+// The render node that will render the custom phase
+#[derive(Default)]
+pub struct CustomNode;
+impl CustomNode {
+    const NAME: &'static str = "custom_node";
+}
+
+impl ViewNode for CustomNode {
+    type ViewData = (
+        &'static RenderPhase<CustomPhaseItem>,
+        &'static ViewTarget,
+        &'static ViewDepthTexture,
+    );
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        (custom_phase, target, depth): QueryItem<Self::ViewData>,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        if custom_phase.items.is_empty() {
+            return Ok(());
+        }
+
+        println!("node running");
```

Comment:
this is done in other examples too, but why are we using `println` instead of tracing's `info`?

Reply:
This is a _very_ work in progress PR and this was just used for debugging. It will definitely not be in the final PR.